kolibri/core/errorreports/api.py

```diff
@@ -8,34 +9,30 @@
 from .models import ErrorReports
 from .serializers import ErrorReportSerializer
 
+
 logger = logging.getLogger(__name__)
 
 
 @api_view(["POST"])
 def report(request):
-    """
-    test with:
-    curl -X POST http://localhost:8000/api/errorreports/report/      -H "Content-Type: application/json"      -d '{
-           "error_message": "An example error occurred",
-           "traceback": "Traceback (most recent call last): ..."
-         }'
-    """
     serializer = ErrorReportSerializer(data=request.data)
     if serializer.is_valid():
         data = serializer.validated_data
         try:
             error = ErrorReports.insert_or_update_error(
-                error_from=FRONTEND,
-                error_message=data["error_message"],
-                traceback=data["traceback"],
+                FRONTEND,
+                data["error_message"],
+                data["traceback"],
+                context=data["context"],
             )
             return Response(
                 {"error_id": error.id if error else None}, status=status.HTTP_200_OK
             )
-        except (AttributeError, Exception) as e:
+
+        except (AttributeError, ValidationError, Exception) as e:
             logger.error("Error while saving error report: {}".format(e))
             return Response(
-                {"error": "Error while saving error report"},
+                {"error": "Error while saving error report: {}".format(e)},
                 status=status.HTTP_500_INTERNAL_SERVER_ERROR,
```

Comment:
Sorry, I had missed this previously - we should never deliberately raise a 500 error, it indicates that something has gone wrong and is unhandled. Instead, this should be a 400 error.